python/lsst/cp/testing/cp.py

```diff
@@ -0,0 +1,190 @@
+           "CptPhotodiodeCorrectionTask", "CptPhotodiodeCorrectionConfig",
+           ]
+
+
+import lsst.cp.pipe as cpPipe
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+
+class CptExtractPtcTaskConnections(cpPipe.ptc.cpExtractPtcTask.PhotonTransferCurveExtractConnections):
+    pass
+
+
+class CptExtractPtcTaskConfig(cpPipe.ptc.PhotonTransferCurveExtractConfig,
+                              pipelineConnections=CptExtractPtcTaskConnections):
+    pass
+
+
+class CptExtractPtcTask(cpPipe.ptc.PhotonTransferCurveExtractTask):
+    configClass = CptExtractPtcTaskConfig
+    _DefaultName = 'cptExtractPtc'
+
+    pass
+
+
+class CptBrighterFatterKernelSolveConnections(pipeBase.PipelineTaskConnections,
+                                              dimensions=("instrument", "detector")):
+    dummy = cT.Input(
+        name="raw",
+        doc="Dummy exposure.",
+        storageClass='Exposure',
+        dimensions=("instrument", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Camera Geometry definition.",
+        storageClass="Camera",
+        dimensions=("instrument", ),
+        isCalibration=True,
+    )
+    inputPtc = cT.Input(
+        name="cptPtc",
+        doc="Input PTC dataset.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+
+    outputBFK = cT.Output(
+        name="cptBrighterFatterKernel",
+        doc="Output measured brighter-fatter kernel.",
+        storageClass="BrighterFatterKernel",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+
+
+class CptBrighterFatterKernelSolveConfig(cpPipe.BrighterFatterKernelSolveConfig,
+                                         pipelineConnections=CptBrighterFatterKernelSolveConnections):
+    pass
+
+
+class CptBrighterFatterKernelSolveTask(cpPipe.BrighterFatterKernelSolveTask):
+
+    ConfigClass = CptBrighterFatterKernelSolveConfig
+    _DefaultName = "cptBfkTask"
+
+    pass
+
+
+class CptLinearitySolveConnections(pipeBase.PipelineTaskConnections,
+                                   dimensions=("instrument", "detector")):
+    dummy = cT.Input(
+        name="raw",
+        doc="Dummy exposure.",
+        storageClass='Exposure',
+        dimensions=("instrument", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Camera Geometry definition.",
+        storageClass="Camera",
+        dimensions=("instrument", ),
+        isCalibration=True,
+    )
+    inputPtc = cT.Input(
+        name="ptc",
+        doc="Input PTC dataset.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+    inputPhotodiodeData = cT.PrerequisiteInput(
```

Comment:
So this should (a) be a regular connection not a prerequisite, and (b) should be part of the ptc extract, not the linearity (so it can be more easily tracked).  See eg https://github.com/lsst/cp_pipe/blob/857a3418e48f9632ccb37132297c3e738fe5e5cb/python/lsst/cp/pipe/ptc/cpExtractPtcTask.py#L54

Reply:
The connection here has been deprecated, I believe.